tests/json.test.js

```diff
@@ -0,0 +1,95 @@
+const t = require("ava");
+const fs = require("fs-extra");
+const path = require("path");
+
+const requiredFields = {
+    owner: "object",
+    record: "object"
+};
+
+const optionalFields = {
+    proxied: "boolean",
+    reserved: "boolean"
+};
+
+const optionalOwnerFields = {
+    email: "string"
+};
+
+const emailRegex = /^[a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,}$/;
+const hostnameRegex = /^(?=.{1,253}$)(?:(?:[_a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?)\.)+[a-zA-Z]{2,63}$/;
+
+const domainsPath = path.resolve("domains");
+const files = fs.readdirSync(domainsPath);
+
+function validateRequiredFields(t, obj, requiredFields, file) {
+    Object.keys(requiredFields).forEach((key) => {
+        t.true(obj.hasOwnProperty(key), `${file}: Missing required field: ${key}`);
+        t.is(typeof obj[key], requiredFields[key], `${file}: Field ${key} should be of type ${requiredFields[key]}`);
+    });
+}
+
+function validateOptionalFields(t, obj, optionalFields, file) {
+    Object.keys(optionalFields).forEach((key) => {
+        if (obj.hasOwnProperty(key)) {
+            t.is(
+                typeof obj[key],
+                optionalFields[key],
+                `${file}: Field ${key} should be of type ${optionalFields[key]}`
+            );
+        }
+    });
+}
+
+t("All files should be valid JSON", (t) => {
+    files.forEach((file) => {
+        t.notThrows(() => fs.readJsonSync(path.join(domainsPath, file)), `${file}: Invalid JSON file`);
+    });
+});
+
+t("All files should have valid file names", (t) => {
+    files.forEach((file) => {
+        t.true(file.endsWith(".json"), `${file}: File does not have .json extension`);
+        t.false(file.includes(".is-a-fullstack.dev"), `${file}: File name should not contain .is-a-fullstack.dev`);
+        t.false(file.includes(".is-cool.dev"), `${file}: File name should not contain .is-cool.dev`);
+        t.false(file.includes(".is-local.org"), `${file}: File name should not contain .is-local.org`);
+        t.false(file.includes(".is-not-a.dev"), `${file}: File name should not contain .is-not-a.dev`);
+        t.false(file.includes(".localplayer.dev"), `${file}: File name should not contain .localplayer.dev`);
+        t.true(file === file.toLowerCase(), `${file}: File name should be lowercase`);
+
+        // Ignore root domain
+        if (file !== "@.json") {
```

Comment:
This should just be the file name for the domain like `is-a-fullstack.dev.json`.

Reply:
So the root domain should be all the domains? Or be everything thats in the `reserved` folder?